playwright/fixtures/common.ts

```diff
@@ -0,0 +1,100 @@
+import { _electron as electron } from "playwright";
+import { test as base, expect } from "@playwright/test";
+import * as fs from "fs";
+import * as path from "path";
+import * as crypto from 'crypto';
+
+export function generateUUID(): string {
+  return crypto.randomBytes(16).toString('hex');
+}
+
+type TestFixtures = {
+  lang: String;
+  theme: String;
+  userdata: String;
+  env: Object;
+  page: any;
+};
+
+const test = base.extend<TestFixtures>({
+  userdata: undefined,
+  env: undefined,
+  lang: "en-US",
+  theme: "light",
+  page: async ({ lang, theme, userdata, env }, use) => {
+    // create userdata path
+    const userDataPathKey = generateUUID();
+    const userDataPath = path.join(__dirname, "../artifacts/userdata", userDataPathKey);
+    fs.mkdirSync(userDataPath, { recursive: true });
+
+    if (userdata) {
+      const userDataFile = path.resolve("playwright/userdata/", `${userdata}.json`);
+      fs.copyFileSync(userDataFile, `${userDataPath}/app.json`);
+    }
+
+    // default environment variables
+    env = Object.assign(
+      {
+        MOCK: true,
+        HIDE_DEBUG_MOCK: true,
+        CI: process.env.CI || undefined,
+        SPECTRON_RUN: true,
```

Comment:
What's `SPECTRON_RUN` doing?

Reply:
it's a env we use inside the app to change the behavior of certain components (like for example, turning off the animation when running the UI tests)

Reply:
Ah gotcha. I guess we should change the name once we've switched but good for now :)